src/mentionui.js

```diff
@@ -0,0 +1,493 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module mention/mentionui
+ */
+
+import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+import ButtonView from '@ckeditor/ckeditor5-ui/src/button/buttonview';
+import Collection from '@ckeditor/ckeditor5-utils/src/collection';
+import BalloonPanelView from '@ckeditor/ckeditor5-ui/src/panel/balloon/balloonpanelview';
+import clickOutsideHandler from '@ckeditor/ckeditor5-ui/src/bindings/clickoutsidehandler';
+import { keyCodes } from '@ckeditor/ckeditor5-utils/src/keyboard';
+import Rect from '@ckeditor/ckeditor5-utils/src/dom/rect';
+
+import TextWatcher from './textwatcher';
+
+import MentionsView from './ui/mentionsview';
+import DomWrapperView from './ui/domwrapperview';
+import MentionListItemView from './ui/mentionlistitemview';
+
+const VERTICAL_SPACING = 5;
+
+/**
+ * The mention UI feature.
+ *
+ * @extends module:core/plugin~Plugin
+ */
+export default class MentionUI extends Plugin {
+	/**
+	 * @inheritDoc
+	 */
+	static get pluginName() {
+		return 'MentionUI';
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	constructor( editor ) {
+		super( editor );
+
+		/**
+		 * The balloon panel view, containing the mention view.
+		 *
+		 * @type {module:ui/panel/balloon/balloonpanelview~BalloonPanelView}
+		 */
+		this.panelView = this._creatPanelView();
+
+		/**
+		 * The mentions view.
+		 *
+		 * @type {module:mention/ui/mentionsview~MentionsView}
+		 * @private
+		 */
+		this._mentionsView = this._createMentionView();
+
+		/**
+		 * Stores mentions feeds configurations.
+		 *
+		 * @type {Map<String, Object>}
+		 * @private
+		 */
+		this._mentionsConfigurations = new Map();
+
+		editor.config.define( 'mention', [] );
+	}
+
+	/**
+	 * @inheritDoc
+	 */
+	init() {
+		// Key listener that handles navigation in mention view.
+		this.editor.editing.view.document.on( 'keydown', ( evt, data ) => {
+			if ( isHandledKey( data.keyCode ) && this.panelView.isVisible ) {
```

Comment:
I'm OCDish but I'd negate this condition to return early and outdent the rest of the stuff that follows.